### DISCUSSION TITLE
Two rigidbodies collide, one gets shot to the side

### DISCUSSION THREAD

**Discussion by LittleCodingFox** [jrouwe/JoltPhysics#750] [Q&A] [answered]
Hello,
I'm making a game engine using Jolt and I did some changes to my code and now for some reason my bodies seem to get shot on collision.

I thought it might be a bug on my code but after reviewing my code multiple times I can't seem to figure out what might cause this, so I'm hoping you might be able to enlighten me as to what could cause this, if that's okay.

I have two bodies, one that is controlled by the player, and another that's not. They're supposed to collide with each other but not move, which was the behaviour thus far. Their settings are:

* Gravity Factor 0
* Dynamic motion
* Allowed DOFs: TranslationX/Y, 2D Plane

So the behaviour is as soon as the player controlled body interacts with the other body, it'll get thrown to the opposite direction at a high speed. I've already checked that setting the body to Static solves the problem, but I'd like it to be dynamic.

I would love it if you could give me some hints on what might be causing this, and I do apologise for making an issue about it rather than fix it myself, but I've already spent hours trying to figure this out and I could use some hints.

Thank you!

**Comment by jrouwe** [comment]
Hello,

I don't know what it is given the info in this ticket. Can you maybe make a repro case by modifying one of the tests in the Samples to show what is going wrong, e.g. by replacing the code [here](https://github.com/jrouwe/JoltPhysics/blob/4e038d190b7e97df01e6a81c8f74551fe487a09d/Samples/Tests/General/SimpleTest.cpp#L25-L46)?

**Comment by LittleCodingFox** [comment]
Hello, it seems that this was caused by me not making one of the bodies static.

However, I think it's strange that two dynamics colliding sends one moving in the opposite side, never really stopping. Is this normal in a zero-gravity situation, when there's no walls or floors?

Thank you very much for your time.

**Comment by jrouwe** [comment, marked as answer]
It's hard to say if what you're seeing is normal or not since you didn't provide a video or repro. In general Newtons first law says that things will keep the same velocity in absence of external forces, so yeah an object should keep moving forever in the same direction after collision in zero g conditions.

**Reply by LittleCodingFox** [reply]
Thank you very much! That must be it! 
Have a great day!
